### PR TITLE
[CARBONDATA-1984][Compression Codec] Double Compression Codec Rectification.

### PR DESCRIPTION
**Pull request by sounakr**
Double datatype compression done through AdaptiveFloatingDelta AdaptiveFloating codec is not returning expected output. These two codec for double compression multiple double with another double. Double multipled with double doesnt give exact output, 1.2 ie one-and-one-fifth can't be exactly represented as a binary fraction. You get close with one-and-an-eigth-and-a-sixteenth.... but never exactly one-fifth. 

So due to the multiply and substract to encode the data, there are some loss of precision and we give a different output when encoded. 

 - [x] Any interfaces changed? No
 
 - [x] Any backward compatibility impacted? No
 
 - [x] Document update required? No

 - [x] Testing done Yes
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. NA

### CODE REVIEW

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/allqueries/AllDataTypesTestCase.scala [hunk 1]
```diff
@@ -1154,4 +1154,114 @@ class AllDataTypesTestCase extends QueryTest with BeforeAndAfterAll {
 
   }
 
+  test("Double Datatype Check with AdaptiveDeltaFloating Codec (BYTE)") {
```
Comment (gvramana) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/allqueries/AllDataTypesTestCase.scala]: Move these tests to new suite

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2545/

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/allqueries/AllDataTypesTestCase.scala (hunk 1 above)
Comment (sounakr) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/allqueries/AllDataTypesTestCase.scala]: Done

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1318/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2555/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2719/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2725/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**